.github/workflows/get_runner_image.yaml

```diff
@@ -0,0 +1,21 @@
+name: Get runner image
+
+on:
+  workflow_call:
+    outputs:
+      runs-on:
+        description: "Image runner"
+        value: ${{ jobs.get-runner-image.outputs.runs-on }}
+
+jobs:
+  get-runner-image:
+    name: Get runner image
+    runs-on: ubuntu-22.04
+    outputs:
+      runs-on: ${{ env.BUILD_ON }}
+    steps:
+      - uses: actions/checkout@v3
+      - name: Get build-on and run-on values
+        run: |
+          echo "BUILD_ON=$(yq '[.bases.[].build-on.[].name, .bases.[].build-on.[].channel] | join("-")' charmcraft.yaml)" >> $GITHUB_ENV
```

Comment:
You can specify multiple `build-on` targets in `charmcraft.yaml`, so I think we should adjust this to:

```yq '[.bases.[].build-on.[0].name, .bases.[].build-on.[0].channel] | join("-")' charmcraft.yaml```